packages/blade/src/components/Tooltip/Tooltip.native.tsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+import { arrow, shift, useFloating, flip, offset } from '@floating-ui/react-native';
+import React from 'react';
+import { Modal, TouchableOpacity } from 'react-native';
+import { TooltipArrow } from './TooltipArrowNative';
+import { TooltipContent } from './TooltipContent';
+import { TooltipProps } from './types';
+import { ARROW_HEIGHT, ARROW_WIDTH } from './constants';
+import { getPlacementParts } from './utils';
+import { useTheme } from '~components/BladeProvider';
+
+const Tooltip = ({
+  content,
+  children,
+  placement = 'left',
+  onOpenChange,
+}: TooltipProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const [isOpen, setIsOpen] = React.useState(false);
+
```

Comment:
isn't a lot of pieces common in this and web implementation? and a lot of calculation parts and functions. possible to keep them at one central place? for eg: if we decide to tomorrow update the padding dev might forget update at both the places and things will become inconsistent

Reply:
The overall structure might look very similar but web & native works quite differently in behaviour. 

In native it works like a Modal, while on web it's a proper mouse over tooltip, so it's best to keep things separated for now.

> if we decide to tomorrow update the padding dev

This file in particular doesn't have any styles which are common across platforms, all the places where we have styles I made sure to only have 1 abstracted out component. eg `TooltipContentWrapper`